### DISCUSSION TITLE
srsUE TDD

### DISCUSSION THREAD

**Discussion by improperaffo** [srsran/srsRAN_Project#382] [General Help] [answered]
Dear all,
We are trying to use srsUE for testing the latency of our setup. We are mainly interested in band 78 (TDD), we know srsUE does not support `30 kHz` scs and band 78 does not currently support `15 kHz` scs, so we tried using band 41, yet we were not able to connect the UE to the gNB either.

Has anyone successfully connected srsUE to srs gNB in TDD mode?

Thanks

**Comment by herlesupreeth** [comment, marked as answer]
band 41 is also TDD and mainly 30khz SCS and srsUE does not support TDD in 5G SA

**Comment by rainey19** [comment]
I've used N41 with srsUE before, but only using ZMQ. Give me an hour or two and I can let you know how it works over RF.

**Reply by improperaffo** [reply]
Thank you. We are interested on ZMQ for now, could you share which configuration options you used?

**Reply by rainey19** [reply]
I've attached the configs I used.

Just realized it requires a code change. In the file lib/src/phy/common/phy_common_nr.c, make the change as follows
<img width="510" alt="Screenshot 2023-12-22 at 3 57 40 PM" src="https://github.com/srsran/srsRAN_Project/assets/19126116/57095c59-eb89-4603-887c-25e888e0797f">

It's probably not the best way to make this change, but it will at least get you as far as msg3 in the RA procedure. 

[n41 configs.zip](https://github.com/srsran/srsRAN_Project/files/13756684/n41.configs.zip)

**Reply by sachin4904** [reply]
Did this work with ZMQ? it got stuck at msg3 for me. @rainey19 @improperaffo

**Reply by pgawlowicz** [reply]
srsUE does not support TDD

**Reply by Zhang-xzhe** [reply]
Does srsUE currently support TDD? I saw this following but I can still not setup.

<img width="1477" height="824" alt="Screenshot from 2026-01-17 14-37-47" src="https://github.com/user-attachments/assets/c7ccd4cc-c407-4eeb-9a43-3f92ac93a065" />

**Reply by pgawlowicz** [reply]
only in LTE mode.
